### PR TITLE
CORE-980 Access Control Documentation

### PR DESCRIPTION
**Pull request by tombusby**
This change is [Reviewable](https://reviewable.io/reviews/scrive/scrive-apidocs/64)

### CODE REVIEW

File: documentation/api/usergroup/deleteusergroup.yaml [hunk 1]
```diff
@@ -42,8 +42,6 @@ post:
           type: string
         action:
           type: string
-  parent_id:
-    type: integer
```
Comment (tombusby) [inline documentation/api/usergroup/deleteusergroup.yaml]: This is a mistake that slipped through from User Group API Doc stuff. It was causing the 403 response not to render.

**Comment by tombusby** [comment]
@claasahl / @wwwater: Could I ask one or both of you guys to check my swagger defs to make sure they're alright.

**Comment by claasahl** [comment]
> @claasahl / @wwwater: Could I ask one or both of you guys to check my swagger defs to make sure they're alright.

@tombusby I'll have a look

**Comment by tombusby** [comment]
@claasahl, cheers mate.

My main concern is that `allowed_actions` will usually have a subset of the properties listed here.

Perhaps I should add `required: []` or something similar.

File: documentation/api/accesscontrol/createrole.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+post:
+  summary: Grant Access Role
+
+  tags:
+    - Access Control
+
+  description: |
+    > Example of JSON that could be used to grant a new role:
+
+    ```json
```
Comment (claasahl) [inline documentation/api/accesscontrol/createrole.yaml]: Does this endpoint allow to set the field `allowed_actions`?

Comment (tombusby) [reply documentation/api/accesscontrol/createrole.yaml]: No, it'll fail if you try to.

File: documentation/definitions/AccessRole.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  }
+description: |
+  JSON representation of an Access Role.
+properties:
+  id:
+    type: integer
+  is_generated:
+    type: boolean
+  role_type:
+    type: string
```
[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/definitions/AccessRole.yaml]: It is, will update.

File: documentation/definitions/AccessRole.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  source:
+    type: object
+    properties:
+      type:
+        type: string
```
[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/definitions/AccessRole.yaml]: It is, will update

File: documentation/definitions/AccessRole.yaml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      id:
+        type: integer
+  target:
+    type: object
+    properties:
+      type:
+        type: string
```
[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/definitions/AccessRole.yaml]: It is, will update

File: documentation/definitions/AccessRole.yaml [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+      - user_group_admin
+      - document_admin
+  source:
+    type: object
+    properties:
+      type:
+        type: string
+        enum:
+          - user
+          - user_group
```
Comment (claasahl) [inline documentation/definitions/AccessRole.yaml]: Should this enum include `folder` (i.e. just like `target.type`)?

Comment (tombusby) [reply documentation/definitions/AccessRole.yaml]: No, only users and usergroups (i.e. the users in the usergroup) can perform actions, so only they can be source.

**Pull request merged**